test/e2e/proxy_test.go

```diff
@@ -95,45 +107,72 @@ func skipIfKubeProxyEnabled(t *testing.T, data *TestData) {
 }
 
 func probeFromNode(node string, url string) error {
-	_, _, _, err := RunCommandOnNode(node, fmt.Sprintf("curl --connect-timeout 1 --retry 5 --retry-connrefused %s", url))
+	_, err := probeFromNodeWithEndpoint(node, url, emptyEp)
 	return err
 }
 
-func probeHostnameFromNode(node string, baseUrl string) (string, error) {
-	url := fmt.Sprintf("%s/%s", baseUrl, "hostname")
-	_, hostname, _, err := RunCommandOnNode(node, fmt.Sprintf("curl --connect-timeout 1 --retry 5 --retry-connrefused %s", url))
-	return hostname, err
-}
-
-func probeClientIPFromNode(node string, baseUrl string) (string, error) {
-	url := fmt.Sprintf("%s/%s", baseUrl, "clientip")
-	_, hostPort, _, err := RunCommandOnNode(node, fmt.Sprintf("curl --connect-timeout 1 --retry 5 --retry-connrefused %s", url))
+func probeFromNodeWithEndpoint(node string, baseURL string, endpoint agnhostEndpoint) (string, error) {
+	url := fmt.Sprintf("%s/%s", baseURL, endpoint)
+	rc, stdout, stderr, err := RunCommandOnNode(node, fmt.Sprintf("curl --connect-timeout 5 --retry 5 --retry-connrefused %s", url))
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("rc: %d, stdout: %s, stderr: %s, err: %v", rc, stdout, stderr, err)
 	}
-	host, _, err := net.SplitHostPort(hostPort)
-	return host, err
+	if endpoint == clientIPEp {
+		host, _, err := net.SplitHostPort(stdout)
+		return host, err
+	}
+	return stdout, nil
```

Comment:
I don't think it is a great design to have a function that processes the output differently based on the endpoint. It may be better IMO to have multiple functions: `probeClientIPEndpointFromNode`, `probeFromNode`, `probeClientIPEndpointFromPod`, `probeFromPod`. These 4 functions can call helper functions (`probeEndpointFromNode` / `probeEndpointFromPod`), but the endpoint-specific output parsing won't happen in the helpers.

Reply:
Indeed. Updated.